### ISSUE TITLE
CentOS - Intel Xeon E3 1225 v3 - I217-LM epserver unable to start err=-22, port=0

### ISSUE THREAD

**Issue by ng5** [eunyoung14/mtcp#123]
Hi,

I'm kind of stuck on below problem from last few days. I'm able to run dpdk sample applications, but not able to make progress with mTCP epserver. I will really appreciate if someone can guide me on below error. this is dell T20 server with only 1 NIC, I believe dual NICs is not a mandatory requirement, but I may be wrong. 
thanks a lot in advance. 
Regards

[root@localhost mtcp]# dpdk-16.11/tools/dpdk-devbind.py --status

Network devices using DPDK-compatible driver
============================================
0000:00:19.0 'Ethernet Connection I217-LM' drv=igb_uio unused=e1000e
Network devices using kernel driver
===================================
<none>
Other network devices
=====================
<none>
Crypto devices using DPDK-compatible driver
===========================================
<none>
Crypto devices using kernel driver
==================================
<none>
Other crypto devices
====================
<none>

[root@localhost example]#  ./epserver -p www_home/ -f epserver-master.conf 
Loading mtcp configuration from : epserver-master.conf
Loading multi-process configuration
device== em1
Loading interface setting
EAL: Detected 4 lcore(s)
EAL: Auto-detected process type: PRIMARY
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
EAL: PCI device 0000:00:19.0 on NUMA socket -1
EAL:   probe driver: 8086:153a net_e1000_em
Total number of attached devices: 1
Interface name: em1
Configurations:
Number of CPU cores available: 4
Number of CPU cores to use: 4
Maximum number of concurrency per core: 10000
Multi-process support is enabled and current core is: 0
Current core is master (for multi-process)
Maximum number of preallocated buffers per core: 10000
Receive buffer size: 8192
Send buffer size: 8192
TCP timeout seconds: 30
TCP timewait seconds: 0
NICs to print statistics: em1
---------------------------------------------------------------------------------
Interfaces:
name: em1, ifindex: 0, hwaddr: 18:66:DA:12:26:02, ipaddr: 192.168.0.5, netmask: 255.255.255.0
Number of NIC queues: 4
---------------------------------------------------------------------------------
Loading routing configurations from : config/route.conf
Routes:
Destination: 192.168.0.0/24, Mask: 255.255.255.0, Masked: 192.168.0.0, Route: ifdx-0
Destination: 192.168.0.5/32, Mask: 255.255.255.255, Masked: 192.168.0.5, Route: ifdx-0
---------------------------------------------------------------------------------
Loading ARP table from : config/arp.conf
ARP Table:
IP addr: 192.168.0.5, dst_hwaddr: 18:66:DA:12:26:02
---------------------------------------------------------------------------------
Initializing port 0... EAL: Error - exiting with code: 1
  Cause: Cannot configure device: err=-22, port=0

**Comment by ajamshed** [comment]
Hi ng5,

Apologies for the delayed response. I saw the log above. It seems that you are using ``em1`` as your dpdk-registered interface. The mTCP stack assumes that your interface name begin with ``dpdk`` (e.g. ``dpdk0``, ``dpdk1``). Some OSes change the alias of the interface after installing the driver. You can rename the interface with the following command:

``$ sudo ip link set em0 name dpdk0``
``$ sudo ip link set em1 name dpdk1``

Can you also give me the output of the following command:
``$ ifconfig``

Thanks.

**Comment by ng5** [comment]
thank you for your time to look into this issue. I tried really hard to get this thing to work, even looked inside source code to add debug printf. I changed all relevant config files (epserver_config, arp & route.conf) and also changed setup_iface.sh etc to change dpdk0,1,2 to network interface i had installed(em1). But I will send the output this evening, I don't have access to my server from work place.  1 quick question - I understand this interface will only be available for DPDK enabled application, so I'm not expecting to be able to browser the internet using this interface. So ideally should I have dual NIC server so that server can be accessed on other device using (SSH etc) and 2nd device will be dedicate to DPDK enabled applications. (My goal is to build custom HTTP(c/c++) server using mTCP). thanks a lot.
